src/Artsy/Analytics/Schema/index.ts

```diff
@@ -7,24 +7,32 @@
  *   `ContextPage` and `Result`.
  */
 
-export * from "./ContextPage"
-export * from "./ContextModule"
-export * from "./Interaction"
-export * from "./Result"
+// TODO: Do we need to export these?
+// export * from "./ContextPage"
+// export * from "./ContextModule"
+// export * from "./Interaction"
+// export * from "./Result"
+
 export * from "./Values"
 
 import { ContextModule } from "./ContextModule"
 import { ContextPage } from "./ContextPage"
+import { Flow } from "./Flow"
 import { AuthenticationInteraction, Interaction } from "./Interaction"
+import { Label } from "./Label"
 import { Failure, Success } from "./Result"
+import { Type } from "./Type"
 
 export type Trackables =
   | AuthenticationInteraction
   | ContextModule
   | ContextPage
+  | Flow
   | Interaction
+  | Label
   | Success
   | Failure
+  | Type
```

Comment:
I was a bit unsure about these additions but it _seemed_ correct. These values should map 1:1 to the segment schema, right? 

so 

```js
context_module: 'Biography',
flow: 'Artwork about the artist',
type: 'Button',
label: 'Read more',
```
-> 
```
```js
context_module: ContextModule.Biography
flow: Flow.ArtworkAboutTheArtist,
type: Type.Button,
label: Label.ReadMore
```

And as more segment fields are added we'll update this similar to how I have above?

Reply:
Would it be possible to dumb this down a bit and create a `Fields` interface that lists out segment properties? Then we can associate those properties with their associated enums in Values. It seems like starting from there we could lock things down further in the future if need be and the system becomes a little less confusing while we're still getting started and defining it.

Reply:
The reason these shapes have separate interfaces is so you only use the right set of properties together. See e.g. https://github.com/artsy/reaction/blob/master/src/Artsy/Analytics/Schema/Interaction.ts, these are the shapes that either a `Interaction` or a `AuthenticationInteraction` should have.

Your event looks to be an `Interaction` which doesn’t have `flow` yet, so if the analytics team is sure that they could potentially see `flow` props on any interaction event then it should be added there.

Reply:
(We can follow up in another PR)